### DISCUSSION TITLE
Calculate the min distance between two grains?

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#1861] [Ask Anything]
Hi, everyone,
     I try to get the min distance between two grains. 
     I know it is easy to get the central location of the grain, but what I want is the distance between two most adjacent pixels

![image](https://github.com/mtex-toolbox/mtex/assets/69452222/f95dd76d-fd60-4a38-8213-dd304fe4fa4c)

**Comment by kilir** [comment]
Hi,
those points must be points on the boundary.
So here's one way:
```
mtexdata titanium

grains = ebsd.calcGrains;


i=36
j=29

gbi = grains(i).boundary;
gbj = grains(j).boundary;
plot([gbi gbj],'coordinates','on')

diffx = repmat(gbi.x,1,length(gbj)) - repmat(gbj.x',length(gbi),1)
diffy = repmat(gbi.y,1,length(gbj)) - repmat(gbj.y',length(gbi),1)

dist = sqrt(diffx.^2 + diffy.^2)
min(dist(:))
```

Cheers,
Rüdiger

**Reply by Brian-Liu-rx** [reply]
Thanks Rüdiger, I will try this code!
Best regards,

Brian Liu
